jOOQ/src/main/java/org/jooq/tools/LoggerListener.java

```diff
@@ -82,91 +84,121 @@
+            if (log.isEnabled(executeLoggingBindVariables)) {
                 String inlined = DSL.using(configuration).renderInlined(ctx.query());
                 if (!ctx.sql().equals(inlined))
-                    log.debug("-> with bind values", newline + inlined);
+                    log.log("-> with bind values", newline + inlined, executeLoggingBindVariables);
             }
+        }
+
+        // [#2987] Log routines
+        else if (log.isEnabled(executeLoggingRoutine) && ctx.routine() != null) {
 
-            // [#2987] Log routines
-            else if (ctx.routine() != null) {
-                log.debug("Calling routine", newline + ctx.sql());
+            log.log("Calling routine", newline + ctx.sql(), executeLoggingRoutine);
 
+            if (log.isEnabled(executeLoggingBindVariables)) {
                 String inlined = DSL.using(configuration)
-                                    .renderInlined(ctx.routine());
+                        .renderInlined(ctx.routine());
 
                 if (!ctx.sql().equals(inlined))
-                    log.debug("-> with bind values", newline + inlined);
+                    log.log("-> with bind values", newline + inlined, executeLoggingBindVariables);
             }
+        } else if (log.isEnabled(executeLoggingSqlString) && !StringUtils.isBlank(ctx.sql())) {
 
-            else if (!StringUtils.isBlank(ctx.sql())) {
-
-                // [#1529] Batch queries should be logged specially
-                if (ctx.type() == ExecuteType.BATCH)
-                    log.debug("Executing batch query", newline + ctx.sql());
-                else
-                    log.debug("Executing query", newline + ctx.sql());
-            }
+            // [#1529] Batch queries should be logged specially
+            if (ctx.type() == ExecuteType.BATCH)
+                log.log("Executing batch query", newline + ctx.sql(), executeLoggingSqlString);
+            else
+                log.log("Executing query", newline + ctx.sql(), executeLoggingSqlString);
+        }
 
-            // [#2532] Log a complete BatchMultiple query
-            else if (batchSQL.length > 0) {
-                if (batchSQL[batchSQL.length - 1] != null)
-                    for (String sql : batchSQL)
-                        log.debug("Executing batch query", newline + sql);
-            }
+        // [#2532] Log a complete BatchMultiple query
+        else if (log.isEnabled(executeLoggingSqlString) && batchSQL.length > 0) {
+            if (batchSQL[batchSQL.length - 1] != null)
+                for (String sql : batchSQL)
+                    log.log("Executing batch query", newline + sql, executeLoggingSqlString);
         }
     }
 
     @Override
     public void recordEnd(ExecuteContext ctx) {
-        if (log.isTraceEnabled() && ctx.record() != null)
-            logMultiline("Record fetched", ctx.record().toString(), Level.FINER);
+        if (log.isEnabled(executeLoggingResult) && ctx.record() != null)
```

Comment:
This is too simplistic and rather dangerous! Before, records were usually not logged individually (except on trace level). Now, there's no way to distinguish between logging records and the entire result. This results in way too verbose logging. More flags will be needed.